pkg/scheduler/cache/cache.go

```diff
@@ -506,12 +506,12 @@ func (cache *schedulerCache) removeNodeImageStates(node *v1.Node) {
 			state, ok := cache.imageStates[name]
 			if ok {
 				state.nodes.Delete(node.Name)
-			}
-			if len(state.nodes) == 0 {
-				// Remove the unused image to make sure the length of
-				// imageStates represents the total number of different
-				// images on all nodes
-				delete(cache.imageStates, name)
+				if len(state.nodes) == 0 {
```

Comment:
I think previous line doesn't make much sense now -`state.nodes.Delete(node.Name)` when len(state.nodes) == 0, you can change the order. I'd to do a `!ok` pattern, atleast log something in the !ok block and then in else we can handle this condition.

Reply:
I think there are still some nits and tests to be fixed as a follow-up for https://github.com/kubernetes/kubernetes/pull/65745. Maybe this can be incorporated into that, or I'll fix it as a follow-up later. :+1:

Reply:
@nikhita Thank you for the fix! I agree more tests are needed - we are working on integration and e2e test at https://github.com/kubernetes/kubernetes/pull/64662 - pls let me know if you find any potential cases we might need to cover.